### PR TITLE
refactor(kerberos): throw runtime error for missing kerberos module

### PR DESCRIPTION
**Pull request by mbroadst**
supersedes #368

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -34,27 +34,27 @@ function relayEvents(listener, emitter, events) {
...
 
-try {
-  const kerberos = requireOptional('kerberos');
-  if (kerberos) {
-    Kerberos = kerberos.Kerberos;
-    MongoAuthProcess = kerberos.processes.MongoAuthProcess;
+  try {
+    kerberos = requireOptional('kerberos');
+  } catch (err) {
+    if (err.code !== 'MODULE_NOT_FOUND') {
```
Comment (daprahamian) [inline lib/utils.js]: If we are now deferring requiring `kerberos` until we actually auth with it, then we shouldn't swallow `MODULE_NOT_FOUND`. If the module isn't found, auth should error.

File: lib/utils.js [hunk 2]
```diff
@@ -34,27 +34,27 @@ function relayEvents(listener, emitter, events) {
...
+      throw new Error('The `kerberos` module was not found. Please install it any try again.');
```
Comment (kvwalker) [inline lib/utils.js]: Is this throwing an error for `kerberos not found` for errors not related to `MODULE_NOT_FOUND`?

File: lib/utils.js (hunk 1 above)
Comment (mbroadst) [reply lib/utils.js]: We're swallowing it to control the text of the returned error to be something more user friendly than "module not found"

File: lib/utils.js (hunk 2 above)
Comment (mbroadst) [reply lib/utils.js]: oops, good catch!

**Pull request merged**